lib/kamal/commands/builder/pack.rb

```diff
@@ -0,0 +1,26 @@
+class Kamal::Commands::Builder::Pack < Kamal::Commands::Builder::Base
+  def push
+    combine \
+      pack(:build,
+        config.repository,
+        "--platform", platform,
+        "--builder", pack_builder,
+        buildpacks,
+        "-t", config.absolute_image,
+        "-t", config.latest_image,
+        "--env", "BP_IMAGE_LABELS=service=#{config.service}",
+        *argumentize("--env", secrets, sensitive: true),
+        "--path", build_context),
+      docker(:push, config.absolute_image),
+      docker(:push, config.latest_image)
+  end
+
+  private
+    def platform
+      "linux/#{local_arches.first}"
+    end
+
+    def buildpacks
+      (pack_buildpacks << "paketo-buildpacks/image-labels").map { |buildpack| [ "--buildpack", buildpack ] }
```

Comment:
I've filed https://github.com/buildpacks/pack/issues/2268 for adding support for `--label` to `pack build`, which if/when added would mean the buildpack injection here could be dropped :-)

Reply:
Love this, thank you @edmorley!